spring-cloud-sleuth-stream/src/main/java/org/springframework/cloud/sleuth/stream/ServerPropertiesHostLocator.java

```diff
@@ -80,7 +83,7 @@ private String getAddress() {
 			address = this.serverProperties.getAddress().getHostAddress();
 		}
 		else {
-			address = "127.0.0.1";
+			address = this.localAdressResolver.getLocalIp4AddressAsString();
```

Comment:
how about this instead. if you are looking to test fallback with mocks, make a package protected method here called `getLocalIp4AddressAsString()` or similar.

Since this class isn't final, in your test you can either mock that method or simply extend the class and supply a constant.

ex.
```java
testThing = new ServerPropertiesHostLocator(..){
  String getLocalIp4AddressAsString() {
    return "123.4.5";
  }
};
```